src/GraphQl/Resolver/Factory/CollectionResolverFactory.php

```diff
@@ -106,21 +106,44 @@ public function __invoke(string $resourceClass = null, string $rootClass = null,
                 }
                 $offset = 1 + (int) $after;
             }
+            if (isset($args['before'])) {
+                $before = base64_decode($args['before'], true);
+                if (false === $before) {
+                    throw Error::createLocatedError(sprintf('Cursor %s is invalid', $args['before']), $info->fieldNodes, $info->path);
+                }
+                $offset = (int) $before - $collection->count();
+                $offset = 0 > $offset ? 0 : $offset;
+            }
+            if (isset($args['last']) && !isset($args['before'])) {
+                $offset = $collection->getTotalItems() - $args['last'];
+                $offset = 0 > $offset ? 0 : $offset;
+            }
 
             $data = ['totalCount' => 0.0, 'edges' => [], 'pageInfo' => ['endCursor' => null, 'hasNextPage' => false]];
+
+            $endCursor = null;
+            $collectionLastIndex = 0;
             if ($collection instanceof PaginatorInterface && ($totalItems = $collection->getTotalItems()) > 0) {
-                $data['pageInfo']['endCursor'] = base64_encode((string) ($totalItems - 1));
                 $data['pageInfo']['hasNextPage'] = $collection->getCurrentPage() !== $collection->getLastPage() && (float) $collection->count() === $collection->getItemsPerPage();
+                $data['pageInfo']['startCursor'] = \base64_encode((string) ($offset));
                 $data['totalCount'] = $totalItems;
+                $collectionLastIndex = $collection->count() - 1;
             }
 
             foreach ($collection as $index => $object) {
+                if ($index === $collectionLastIndex) {
```

Comment:
Why is this needed? The `$collectionLastIndex` is always the endCursor doesn't it?

Reply:
You are completly right, it's not needed at all